### PR TITLE
I5852 esc key modal

### LINKED ISSUE #5852: "Esc" key won't close the Delete my profile modal (or any Overlay)
STR:
1. Load AMO dev and go to Edit my profile page
2. Click the Delete my profile button
3. Try to close the modal using "Esc" key

Actual result:
The modal is still displayed.

Expected result:
The modal closes.

Notes:
This issue is reproducible on AMO dev and AMO stage with FF61(Win10).
This is also reproducible for the make a review modal from an add-on/theme's detail page.

Delete my profile
delete my profile

Write a review
write a review

### PR DESCRIPTION
**Pull request by rebmullin**
fixes #5852

### CODE REVIEW

File: src/ui/components/Overlay/index.js [hunk 1]
```diff
@@ -3,16 +3,19 @@ import invariant from 'invariant';
...
 import './styles.scss';
 
+const { ESC } = Keys;
+
 type Props = {|
   className?: string,
   children: React.Element<any>,
   id: string,
-  onEscapeOverlay?: () => void,
+  onEscapeOverlay?: (event: SyntheticEvent<any>) => void,
```
[the remark this answers is not in the document]
Comment (rebmullin) [reply src/ui/components/Overlay/index.js]: yeh I was loving this and I think I'll try your first suggestion.

[the remark this answers is not in the document]
Comment (rebmullin) [reply src/ui/components/Overlay/index.js]: thank you : )

File: src/ui/components/Overlay/index.js [hunk 2]
```diff
@@ -22,10 +25,15 @@ type UIStateType = {|
...
 
+type keyDownType = {|
+  event: Object,
+|};
+
 type InternalProps = {|
   ...Props,
   setUIState: ($Shape<UIStateType>) => void,
   uiState: UIStateType,
+  keydown?: keyDownType | void,
```
[the remark this answers is not in the document]
Comment (rebmullin) [reply src/ui/components/Overlay/index.js]: hmm..that's a good question : ).  I *think* maybe I will make it optional..

basically keydown.event will be null when you open the modal but keydown itself is always defined it seems (keydown.event is defined only when you press they "ESC"). I'll update the event in the keyDownType to reflect this..

File: src/ui/components/Overlay/index.js [hunk 3]
```diff
@@ -22,10 +25,15 @@ type UIStateType = {|
 
 const initialUIState: UIStateType = { visible: false };
 
+type keyDownType = {|
+  event: Object,
```
[the remark this answers is not in the document]
Comment (rebmullin) [reply src/ui/components/Overlay/index.js]: hmm i updated to be: `event: SyntheticEvent<any> | null` - is this ok?

File: src/ui/components/Overlay/index.js [hunk 4]
```diff
@@ -22,10 +25,15 @@ type UIStateType = {|
 
 const initialUIState: UIStateType = { visible: false };
 
+type keyDownType = {|
```
Comment (bobsilverberg) [inline src/ui/components/Overlay/index.js]: My nit from the last iteration stands. I'm not sure you need to define a `keyDownType` when it's only used once, below. You could just define `InternalProps` to include 

`keydown: {| event: SyntheticEvent<any> | null |}`

Comment (rebmullin) [reply src/ui/components/Overlay/index.js]: I missed that...yes I'll combine into InternalProps. That makes sense. thanks

**Comment by rebmullin** [comment]
thanks for the review : )

**Pull request merged**